pkg/server/cloud/aws/instances.go

```diff
@@ -441,6 +450,38 @@ func (e *AwsEC2) retrieveOrAllocateHost(node *api.Node) (*string, error) {
 	return allocateOutput.HostIds[0], nil
 }
 
+func (e *AwsEC2) isHostAvailable(hostId *string) bool {
+	describeOutput, err := e.client.DescribeHosts(&ec2.DescribeHostsInput{
+		HostIds:    []*string{hostId},
+	})
+	if err != nil {
+		klog.Errorf("cannot describe host %s: %v", *hostId, err)
+		return false
+	}
+	if len(describeOutput.Hosts) != 1 {
+		return false
+	}
+	host := describeOutput.Hosts[0]
+	return *host.State == "available"
+}
+
+func (e *AwsEC2) waitForHostAvailable(ctx context.Context, hostId *string) error {
+	hostAvailable := false
+	for {
+		select {
+		case <-ctx.Done():
+			if !hostAvailable {
+				return fmt.Errorf("host not available")
+			} else {
+				return nil
+			}
+		default:
+			hostAvailable = e.isHostAvailable(hostId)
+		}
+		time.Sleep(500 * time.Millisecond)
```

Comment:
Instead of a direct sleep here, you could use a Ticker in the `select` statement above instead of `default`.